source/UF/SizeTruncationandConnectedness.lagda

```diff
@@ -0,0 +1,346 @@
+ Prop-2-2 {𝓤} {𝓦} {A} {X} {f} {succ n} j f-con A-small X-is-loc-small =
+  j A-small
+    (locally-small-from-surjection (−1-connected-maps-are-surjections f-−1-con))
+    f-−1-con
+  where
+   f-−1-con : f is −1 connected-map
+   f-−1-con = map-connectedness-is-lower-closed ⋆ f-con
+   helper : (x x' : X)
+          → Σ a ꞉ A , f a ＝ x
+          → Σ a ꞉ A , f a ＝ x'
+          → (x ＝ x') is 𝓥 small
+   helper .(f a) .(f a') (a , refl) (a' , refl) =
+    Prop-2-2 j (ap-is-less-connected (ua (𝓤 ⊔ 𝓦)) f f-con)
+             (small-implies-locally-small A 𝓥 A-small a a')
+             (X-is-loc-small (f a) (f a')) 
+   locally-small-from-surjection : is-surjection f
+                                 → (x x' : X)
+                                 → (x ＝ x') is 𝓥 small
+   locally-small-from-surjection f-is-surj x x' =
+    ∥∥-rec₂ (being-small-is-prop ua (x ＝ x') 𝓥)
+            (helper x x')
+            (f-is-surj x)
+            (f-is-surj x')
+\end{code}
+
+Lemma 2.3.
+
+\begin{code}
+
+ Lemma-2-3 : {X : 𝓤 ̇} {n : ℕ₋₂}
+           → Propositional-Resizing
+           → X is (n + 1) truncated
+           → X is ι (n + 2) locally-small
+ Lemma-2-3 {_} {X} {−2} pr X-prop =
+  pr X (is-prop'-implies-is-prop X-prop)
+ Lemma-2-3 {_} {_} {succ n} pr X-trunc x x' =
+  Lemma-2-3 pr (X-trunc x x')
+
+\end{code}
+
+Lemma 2.4.
+
+\begin{code}
+
+ Lemma-2-4 : {X : 𝓤 ̇} {Y : 𝓦 ̇} {f : X → Y} {n : ℕ₋₂}
+           → Propositional-Resizing
+           → f is (n + 1) truncated-map
+           → Y is ι (n + 2) locally-small
+           → X is ι (n + 2) locally-small
+ Lemma-2-4 {_} {_} {_} {_} {f} {_} pr f-trunc Y-loc-small =
+  locally-small-closed-under-≃ (total-fiber-is-domain f)
+   (locally-small-closed-under-Σ Y-loc-small (λ y → Lemma-2-3 pr (f-trunc y)))
+
+\end{code}
+
+Lemma 2.5.
+
+\begin{code}
+
+ Lemma-2-5 : {X : 𝓤 ̇} {Y : 𝓦 ̇} {f : X → Y} {n : ℕ₋₂}
+           → Join-Construction-Result {𝓤} {𝓤}
+           → Propositional-Resizing
+           → f is (n + 1) truncated-map
+           → Y is ι (n + 2) locally-small
+           → X is (n + 1) connected
+           → X is 𝓥 small
+ Lemma-2-5 {𝓤} {_} {X} {_} {_} {n} j pr f-trunc Y-loc-small X-conn =
+  ∥∥-rec (being-small-is-prop ua X 𝓥)
+   X-inhabited-implies-small (center X-−1-conn)
+  where
+   X-loc-small : X is ι (n + 2) locally-small
+   X-loc-small = Lemma-2-4 pr f-trunc Y-loc-small
+   X-−1-conn : X is −1 connected
+   X-−1-conn = connectedness-is-lower-closed' ⋆ X-conn
+   X-point : X → 𝟙 {𝓤} → X
+   X-point x ⋆ = x
+   X-point-n-conn : (x : X) → (X-point x) is n connected-map
+   X-point-n-conn x = basepoint-map-is-less-connected (ua 𝓤) (X-point x) X-conn
+   𝟙-is-small : 𝟙 {𝓤} is 𝓥 small
+   𝟙-is-small = pr 𝟙 𝟙-is-prop
+   X-inhabited-implies-small : X → X is 𝓥 small
+   X-inhabited-implies-small x =
+    Prop-2-2 j (X-point-n-conn x) 𝟙-is-small X-loc-small
+
+\end{code}
+
+We shall follow Dan's updated result and prove the following in the absence of
```

Comment:
It will be unclear to the readers other than a few of us what "Dan's updated result" is. I think you need to clarify the story here.